components/org.wso2.carbon.identity.api.server.notification.sender/org.wso2.carbon.identity.api.server.notification.sender.v2/src/main/java/org/wso2/carbon/identity/api/server/notification/sender/v2/impl/NotificationSendersApiServiceImpl.java

```diff
@@ -0,0 +1,221 @@
+    }
+
+    @Override
+    public Response createPushSender(PushSenderAdd pushSenderAdd) {
+
+        PushSender pushSender = notificationSenderManagementService.addPushSender(pushSenderAdd);
+        URI location = null;
+        try {
+            location = ContextLoader.buildURIForHeader(
+                    V2_API_PATH_COMPONENT + NOTIFICATION_SENDER_CONTEXT_PATH + "/" + PUSH_PUBLISHER_TYPE + "/" +
+                            URLEncoder.encode(pushSender.getName(), StandardCharsets.UTF_8.name())
+                                    .replace(PLUS, URL_ENCODED_SPACE));
+        } catch (UnsupportedEncodingException e) {
+            ErrorResponse errorResponse =
+                    new ErrorResponse.Builder().withMessage("Error due to unsupported encoding.").build();
+            throw new APIError(Response.Status.METHOD_NOT_ALLOWED, errorResponse);
+        }
+        return Response.created(location).entity(pushSender).build();
+    }
+
+    @Override
+    public Response createSMSSender(SMSSenderAdd smSSenderAdd) {
+
+        SMSSender smsSender = notificationSenderManagementService.addSMSSender(smSSenderAdd);
+        URI location = null;
+        try {
+            location = ContextLoader.buildURIForHeader(
+                    V2_API_PATH_COMPONENT + NOTIFICATION_SENDER_CONTEXT_PATH + "/" + SMS_PUBLISHER_TYPE + "/" +
+                            URLEncoder.encode(smsSender.getName(), StandardCharsets.UTF_8.name())
+                                    .replace(PLUS, URL_ENCODED_SPACE));
+        } catch (UnsupportedEncodingException e) {
+            ErrorResponse errorResponse =
+                    new ErrorResponse.Builder().withMessage("Error due to unsupported encoding.").build();
+            throw new APIError(Response.Status.METHOD_NOT_ALLOWED, errorResponse);
+        }
+        return Response.created(location).entity(smsSender).build();
+    }
+
+    @Override
+    public Response deleteEmailSender(String senderName) {
+
+        if (StringUtils.equals(getTenantDomainFromContext(), MultitenantConstants.SUPER_TENANT_DOMAIN_NAME)) {
+            return Response.status(Response.Status.METHOD_NOT_ALLOWED).build();
+        }
+        notificationSenderManagementService.deleteNotificationSender(senderName);
+        return Response.noContent().build();
+    }
+
+    @Override
+    public Response deletePushSender(String senderName) {
+
+        notificationSenderManagementService.deleteNotificationSender(senderName);
+        return Response.noContent().build();
+    }
+
+    @Override
+    public Response deleteSMSSender(String senderName) {
+
+        notificationSenderManagementService.deleteNotificationSender(senderName);
+        return Response.noContent().build();
+    }
+
+    @Override
+    public Response getEmailSender(String senderName) {
+
+        if (StringUtils.equals(getTenantDomainFromContext(), MultitenantConstants.SUPER_TENANT_DOMAIN_NAME)) {
```

Comment:
> notificationSenderManagementService.deleteNotificationSender(senderName);

Could have refactor this to a utility method - isSuperTenant ..

Reply:
There is no Util class present in identity-server